### DISCUSSION TITLE
Oceananigans vs MITgcm comparison in the GCM hydrostatic mode

### DISCUSSION THREAD

**Discussion by simone-silvestri** [CliMA/Oceananigans.jl#2896] [Show and tell]
@jm-c and I have recently set up a latitude-longitude global ocean simulation at a 1-degree resolution to compare results from Oceananigans and MITgcm.

The setup is the following:
- 48 vertical levels taken from the ECCO2 version 4 grid
- Horizontal laplacian viscosity and diffusivity with values $\nu_h = 50000 \ m^2/s$ and $\kappa_h = 300 m^2/s$ , respectively
- Vertical laplacian viscosity $\nu_v = 10^{-4} m^2/s$ increased to $10^{-2} m^2/s$ in the top 50 meters
- Vertical laplacian diffusivity $\kappa = 10^{-5} m^2/s$
- Linear bottom drag with coefficient $\mu = 10^{-3}$
- Centered advection schemes for tracers
- Energy-conserving vector invariant momentum advection
- Linear equation of state
- No GM or Redi diffusivities

we evolve for 3 months starting from the same initial condition with the same forcing. The results in terms of $$T^{\text{Oceananigans}} - T^{\text{MITgcm}}$$
is shown below. The left plot shows the surface, while the right shows a transect of the Pacific ocean.

![temperature_mitgcm_minus_oceananigans_at_80days_left_sst_right_at_i_equal_to_10 (2)](https://user-images.githubusercontent.com/33547697/216651589-a4d53900-c9fd-40f7-932a-7de6d0fb878b.png)

**Comment by glwagner** [comment]
The key difference between the two is the treatment of vorticity on horizontal immersed boundaries correct?

**Comment by simone-silvestri** [comment]
Yes. MITgcm sets vorticity to zero on `peripheral_nodes`, while we do not. So we end up having a vorticity at corners (vorticity on flat edges will still be zero)
